### PR TITLE
qemacs: fix cross-compilation

### PR DESCRIPTION
**Pull request by ghost**
###### Description of changes

This fixes cross-compilation for qemacs.  It also adds enableX11 parameter.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] powerpc64le-linux
  - [x] aarch64-linux
  - [x] mips64el-linux (cross from powerpc64le-linux)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Mindavi** [comment]
Build on darwin seems broken, is it related?

**Review by iblech — APPROVED** [review]
Very nice, thank you for your work! I didn't test cross compilation, all the changes look very solid.

I cannot comment on the darwin situation.

**Comment by amjoseph-nixpkgs** [comment]
> Build on darwin seems broken, is it related?

I'm afraid I do not have access to any darwin machines.

**Comment by Artturin** [comment]
> > Build on darwin seems broken, is it related?
> 
> I'm afraid I do not have access to any darwin machines.

you can request access to https://github.com/winterqt/darwin-build-box if you want

File: pkgs/applications/editors/qemacs/default.nix [hunk 1]
```diff
@@ -9,12 +15,42 @@ stdenv.mkDerivation rec {
...
+    "--cross-prefix=${stdenv.cc.targetPrefix}"
+    "--extra-cflags=-Ilibqhtml"
+  ] ++ lib.optionals (!enableX11) [
+    "--disable-x11"
+  ];
+
+  makeFlags = [
+    # is actually used as BUILD_CC
+    "HOST_CC=${buildPackages.stdenv.cc}/bin/cc"
+    "CC=${stdenv.cc.targetPrefix}cc"
```
Comment (Artturin) [inline pkgs/applications/editors/qemacs/default.nix:30]: upstream assumes gcc

File: pkgs/applications/editors/qemacs/default.nix [hunk 2]
```diff
@@ -9,12 +15,42 @@ stdenv.mkDerivation rec {
     sha256 = "156z4wpj49i6j388yjird5qvrph7hz0grb4r44l4jf3q8imadyrg";
   };
 
-  buildInputs = [ libpng libX11 libXext libXv ];
+  buildInputs = lib.optionals enableX11 [ libpng libX11 libXext libXv ];
+
```
Comment (Artturin) [inline pkgs/applications/editors/qemacs/default.nix:19]: removed unneeded `configurePlatforms = [ ];` , the build system just ignores them

Comment (amjoseph-nixpkgs) [reply pkgs/applications/editors/qemacs/default.nix:19]: > removed unneeded `configurePlatforms = [ ];` , the build system just ignores them

It was probably needed for https://github.com/NixOS/nixpkgs/pull/209418; I wrote both PRs and then tried to separate them into two commits.

**Pull request closed without merging**